src/Table/Table.types.ts

```diff
@@ -9,23 +9,17 @@ export interface CellInfoType {
   row: RowType;
 }
 
-interface ColumnInfoType {
-  align?: ColumnAlignment;
-  label: string;
-  dataKey?: Key;
-  width?: string | number;
-}
-
 type ColumnAlignment = "left" | "right" | "center";
 
 export type ColumnType = {
```

Comment:
IMHO we need to offer a generic to express the metadata type lest we end up with scenarios like this:

```typescript
/**
  * assume metadata is... { actionText: "Button clicked" }
  */

const buttonRenderer = ({ label, metadata }: ColumnType) => (
  <Button onClick={action(metadata.actionText)}>{label}</Button> // Argument of type 'unknown' is not assignable to parameter of type 'string'
);
```

Reply:
That's a good idea. I pushed a change.